Mathlib/Probability/Martingale/OptionalSampling.lean

```diff
@@ -142,13 +142,16 @@ and is a measurable space with the Borel σ-algebra. -/
 
 
 variable {ι : Type*} [LinearOrder ι] [LocallyFiniteOrder ι] [OrderBot ι] [TopologicalSpace ι]
-  [DiscreteTopology ι] [MeasurableSpace ι] [BorelSpace ι] [MeasurableSpace E] [BorelSpace E]
+  [OrderTopology ι] [MeasurableSpace ι] [BorelSpace ι] [MeasurableSpace E] [BorelSpace E]
   [SecondCountableTopology E] {ℱ : Filtration ι m} {τ σ : Ω → ι} {f : ι → Ω → E} {i : ι}
 
 theorem condexp_stoppedValue_stopping_time_ae_eq_restrict_le (h : Martingale f ℱ μ)
     (hτ : IsStoppingTime ℱ τ) (hσ : IsStoppingTime ℱ σ) [SigmaFinite (μ.trim hσ.measurableSpace_le)]
     (hτ_le : ∀ x, τ x ≤ i) :
     μ[stoppedValue f τ|hσ.measurableSpace] =ᵐ[μ.restrict {x : Ω | τ x ≤ σ x}] stoppedValue f τ := by
+  have := LinearLocallyFiniteOrder.succOrder ι
+  have := LinearLocallyFiniteOrder.predOrder ι
+  have := discreteTopology_iff_orderTopology_of_pred_succ.mpr ‹_›
```

Comment:
@urkud Could you please explain why you now want the `OrderTopology` assumption, instead of `DiscreteTopology`?

Reply:
I thought that I need them to state the theorems, but I was wrong. Reverted.